### DISCUSSION TITLE
SwiftUI and Realm: path forward

### DISCUSSION THREAD

**Discussion by HKdAlex** [realm/realm-cocoa#8480] [Q&A] [answered]
Is there some place one can read about the ideas/plans for the way forward for Realm and SwiftUI?

E.g., there are some quite obvious performance or similar issues when using Realm and SwiftUI.

Or Apple has released the @Observable macro recently but there is no documentation on how it plays along with Realm in SwiftUI or whether it can be used with Realm at all.

**Comment by nirinchev** [comment, marked as answer]
We don't have a public roadmap for our SwiftUI integration. A lot of the changes/enhancements we work on are a direct result of customer feedback, so if you have a burning need for something, feel free to upvote an existing issue or create a new one. We do periodic reviews of most upvoted issues and incorporate this as a datapoint during our planning process.

Regarding the obvious performance issues, I'm not sure what you mean - I am not aware of any open issues that indicate a performance regression when using SwiftUI compared to UIKit. There are indeed some usability issues, which we do plan to tackle, though we don't have a timeline for each individual one.
